Translators/WZDx/tim_generator.py

```diff
@@ -175,15 +179,35 @@ def get_first_road_name(feature):
 
 
 def get_data_frames(feature):
-    coords = feature["geometry"]["coordinates"]
+    coords = copy.deepcopy(feature["geometry"]["coordinates"])
     anchor = get_anchor(feature)
     if anchor is None:
         return None
     start_date = feature["properties"]["start_date"]
     duration = get_duration_time_minutes(feature)
     msgId = get_msg_id(anchor)
-    regions = [get_region(
-        coords, anchor, roadName=get_first_road_name(feature))]
+
+    # In here need to calculate if size of path is greater than 63
+    # then generate regions accordingly
+    maxLength = 62
+    msgLength = len(feature['geometry']['coordinates'])
+    regions = []
+    if (msgLength > maxLength):
+        numPaths = math.ceil(msgLength / maxLength)
+        pathLength = (int) (msgLength / numPaths)
+        remainder = msgLength % pathLength
+
+        for index in range(0, numPaths):
+            rangeEnd = pathLength + 1 if index < remainder else pathLength
+            regionCoords = feature['geometry']['coordinates'][0 : rangeEnd]
+            anchor = get_anchor(feature)
+            regions.append(get_region(regionCoords, anchor, roadName=get_first_road_name(feature)))
+            del feature['geometry']['coordinates'][0 : rangeEnd]
+    else: 
+        regions = [get_region(
+            coords, anchor, roadName=get_first_road_name(feature))]
+
+    feature["geometry"]["coordinates"] = coords

```

Comment:
Rather than deleting and then resetting the coordinates, could we do our calculations on the deep copied collection?

Reply:
Updated this to do calculations on deep copied collection